backend/core/src/listings/combined/combined-listing-transformer.ts

```diff
@@ -0,0 +1,97 @@
+import { Jurisdiction } from "../../jurisdictions/entities/jurisdiction.entity"
+import { ListingMultiselectQuestion } from "../..//multiselect-question/entities/listing-multiselect-question.entity"
+import { ReservedCommunityType } from "../../reserved-community-type/entities/reserved-community-type.entity"
+import { Address } from "../../shared/entities/address.entity"
+import { mapTo } from "../../shared/mapTo"
+import { Unit } from "../../units/entities/unit.entity"
+import { ListingFeatures } from "../entities/listing-features.entity"
+import { ListingImage } from "../entities/listing-image.entity"
+import { ListingUtilities } from "../entities/listing-utilities.entity"
+import { Listing } from "../entities/listing.entity"
+import { summarizeUnitsByTypeAndRent } from "../../shared/units-transformations"
+
+/**
+ * Converts raw table row from combined_listings view into a Listing object
+ * REMOVE_WHEN_EXTERNAL_NOT_NEEDED
+ */
+export class CombinedListingTransformer {
+  public transformAll(results): Listing[] {
+    return results.map((result) => {
+      return this.transform(result)
+    })
+  }
+
+  public transform(result): Listing {
+    const mapToOpts = { excludeExtraneousValues: true }
+
+    const listing = new Listing()
+    listing.id = result.id
+    listing.assets = result.assets
+    listing.unitsAvailable = result.units_available
+    listing.applicationDueDate = result.application_due_date
+    listing.applicationOpenDate = result.application_open_date
+    listing.name = result.name
+    listing.waitlistCurrentSize = result.waitlist_current_size
+    listing.waitlistMaxSize = result.waitlist_max_size
+    listing.status = result.status
+    listing.reviewOrderType = result.review_order_type
+    listing.publishedAt = result.published_at
+    listing.closedAt = result.closed_at
+    listing.updatedAt = result.updated_at
+    listing.urlSlug = result.url_slug
+    listing.isExternal = result.is_external
+
+    // jurisdiction
+    listing.jurisdiction = mapTo(Jurisdiction, result.jurisdiction)
+
+    // units
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    listing.units = mapTo(Unit, result.units as Array<any>)
```

Comment:
This is a problem we've ran into on Bloom, but the mapping an object to its class can be really memory intensive. What we found was the more units the more likely we were to crash our backend and so we split it up as best it could

I know that you switched to using the View and also manually map things out here, but it may be worth testing what happens when you have a listing with many unit (like 100 units) and seeing how this performs

Reply:
Interesting...  I imagine I could probably iterate through the raw objects directly and map them one at a time rather than passing the array into the mapper, deleting the raw as I go.  That should roughly halve the amount of ram used at peak in exchange for a few more LOC.  That might help if this is specific to the mapping function, but since it really only reduces the peak memory _per listing_, it might not be enough to help reduce total memory usage.  Peak during mapping would be lower for a single listing with 100 units with that approach, for example, but total memory usage for 3 listings with 50 units each would still be higher with or without manual iteration (likewise with serving multiple separate requests concurrently across the process), so we probably want to try to understand the problem better before making any additional changes.

Some questions: 
- The OOM errors you were getting was during the mapping phase specifically, correct?  If so, do you know what was causing it to use so much ram?  Obviously it would be `units * 2` just from having 2 copies of every object being mapped, but is there anything else going on there?
- Did you see the same issue mapping objects other than units? Multiselect questions, etc
- Do you have any numbers on what that memory usage looked like when you were experiencing that before?
- One thing I did notice is that the query as written for the `/listings` endpoint returns roughly `listings * 30` flattened rows with every query (ie 15 listings returns >450 rows), which TypeORM then loops through and groups together into nested relations.  Could this be related to the memory problems you are seeing or is that something else?

Reply:
Resolving based on conversation with Morgan.  This appears to primarily be an issue with the large resultsets created by queries with multiple joins and doesn't necessarily apply to our use case.  It's good info though, and something to circle back on in the future.